index.js

```diff
@@ -4,20 +4,29 @@ var crypto = require('crypto');
 var isStream = require('is-stream');
 var Promise = require('pinkie-promise');
 
-var hasha = module.exports = function (buf, opts) {
+var hasha = module.exports = function (input, opts) {
 	opts = opts || {};
 
-	var inputEncoding = typeof buf === 'string' ? 'utf8' : undefined;
 	var outputEncoding = opts.encoding || 'hex';
 
 	if (outputEncoding === 'buffer') {
 		outputEncoding = undefined;
 	}
 
-	return crypto
-		.createHash(opts.algorithm || 'sha512')
-		.update(buf, inputEncoding)
-		.digest(outputEncoding);
+	var hash = crypto.createHash(opts.algorithm || 'sha512');
+
+	function update(buf) {
```

Comment:
@jamestalmage According to the spec, function declarations inside blocks is a syntax error, but all engines handle that for us by rewriting it. I think they made it official in ES6. Again, no source, but you might be able to Google it up.